dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/Main.java

```diff
@@ -47,6 +48,7 @@ private static void printErrorMessages(Throwable e) {
               ImmutableList<String> errorMessages =
                   Throwables.getCausalChain(e).stream()
                       .map(Throwable::getMessage)
+                      .filter(Objects::nonNull)
```

Comment:
Are you sure? Would you rather toString() the Throwables, message or not? If one of them is null, this will hide the error. Throwable.toString() is probably our friend instead of getMessage().

Reply:
The full `Throwable` is already printed above.
https://github.com/google/dwh-migration-tools/blob/d7613ccdf3ae71c527166233b257c17a370fa93c/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/Main.java#L77

The `printErrorMessages` method provides a quick summary of the error at the end of the dumper output. Such an error summary is emphasized with the stars. This allows faster analysis without the need to manually scan a lot of text above to find the error message.

Sample output:
```
com.zaxxer.hikari.pool.HikariPool$PoolInitializationException: Failed to initialize pool: FATAL: password authentication failed for user "joe"
	at com.zaxxer.hikari.pool.HikariPool.throwPoolInitializationException(HikariPool.java:576)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:562)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:81)
	at com.google.edwmigration.dumper.application.dumper.handle.JdbcHandle.newPooledJdbcHandle(JdbcHandle.java:50)
	at com.google.edwmigration.dumper.application.dumper.connector.redshift.AbstractRedshiftConnector.open(AbstractRedshiftConnector.java:240)
	at com.google.edwmigration.dumper.application.dumper.MetadataDumper.run(MetadataDumper.java:183)
	at com.google.edwmigration.dumper.application.dumper.MetadataDumper.run(MetadataDumper.java:95)
	at com.google.edwmigration.dumper.application.dumper.MetadataDumper.run(MetadataDumper.java:72)
	at com.google.edwmigration.dumper.application.dumper.Main.run(Main.java:40)
	at com.google.edwmigration.dumper.application.dumper.Main.main(Main.java:69)
Caused by: java.sql.SQLException: FATAL: password authentication failed for user "joe"
	at com.amazon.redshift.util.RedshiftException.getSQLException(RedshiftException.java:56)
	at com.amazon.redshift.Driver.connect(Driver.java:326)
	at org.springframework.jdbc.datasource.SimpleDriverDataSource.getConnectionFromDriver(SimpleDriverDataSource.java:144)
	at org.springframework.jdbc.datasource.AbstractDriverBasedDataSource.getConnectionFromDriver(AbstractDriverBasedDataSource.java:205)
	at org.springframework.jdbc.datasource.AbstractDriverBasedDataSource.getConnection(AbstractDriverBasedDataSource.java:169)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:541)
	... 9 more
********************************************************************
* ERROR
* Failed to initialize pool: FATAL: password authentication failed for user "joe"
* Caused by: FATAL: password authentication failed for user "joe"
********************************************************************
```